bundles/org.openhab.binding.zoneminder/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,217 @@
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<bridge-type id="server">
+		<label>ZoneMinder Server</label>
+		<description>Represents a ZoneMinder server</description>
+		<channels>
+			<channel id="imageMonitorId" typeId="id">
+				<label>Image Monitor Id</label>
+				<description>Monitor ID for Image URL Channel</description>
+			</channel>
+			<channel id="imageUrl" typeId="url">
+				<label>Image URL</label>
+			</channel>
+			<channel id="videoMonitorId" typeId="id">
+				<label>Video Monitor Id</label>
+				<description>Monitor ID for Video URL Channel</description>
+			</channel>
+			<channel id="videoUrl" typeId="url">
+				<label>Video URL</label>
+			</channel>
+		</channels>
+		<config-description-ref uri="thing-type:zoneminder:server"/>
+	</bridge-type>
+
+	<thing-type id="monitor">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="server"/>
+		</supported-bridge-type-refs>
+
+		<label>ZoneMinder Monitor</label>
+		<description>Represents a ZoneMinder monitor</description>
+		<channels>
+			<channel id="id" typeId="id"/>
+			<channel id="name" typeId="name"/>
+			<channel id="image" typeId="image"/>
+			<channel id="enable" typeId="enable"/>
+			<channel id="function" typeId="function"/>
+			<channel id="alarm" typeId="alarm"/>
+			<channel id="state" typeId="state"/>
+			<channel id="triggerAlarm" typeId="triggerAlarm"/>
+			<channel id="hourEvents" typeId="events">
+				<label>Hour Events</label>
+			</channel>
+			<channel id="dayEvents" typeId="events">
+				<label>Day Events</label>
+			</channel>
+			<channel id="weekEvents" typeId="events">
+				<label>Week Events</label>
+			</channel>
+			<channel id="monthEvents" typeId="events">
+				<label>Month Events</label>
+			</channel>
+			<channel id="totalEvents" typeId="events">
+				<label>Total Events</label>
+			</channel>
+			<channel id="imageUrl" typeId="url">
+				<label>Image URL</label>
+			</channel>
+			<channel id="videoUrl" typeId="url">
+				<label>Video URL</label>
+			</channel>
+			<channel id="eventId" typeId="eventId"/>
+			<channel id="eventName" typeId="eventName"/>
+			<channel id="eventCause" typeId="eventCause"/>
+			<channel id="eventNotes" typeId="eventNotes"/>
+			<channel id="eventStart" typeId="eventStart"/>
+			<channel id="eventEnd" typeId="eventEnd"/>
+			<channel id="eventFrames" typeId="eventFrames"/>
+			<channel id="eventAlarmFrames" typeId="eventAlarmFrames"/>
+			<channel id="eventLength" typeId="eventLength"/>
+		</channels>
+		<config-description-ref uri="thing-type:zoneminder:monitor"/>
+	</thing-type>
+
+	<channel-type id="id">
+		<item-type>String</item-type>
+		<label>ID</label>
+		<description>Monitor ID</description>
+		<state readOnly="true" pattern="%s"></state>
+	</channel-type>
```

Comment:
Should this be a thing property instead?

Reply:
If you change the monitor id in Zoneminder, you should be able to change it here as well. Otherwise, you would need to delete and readd the thing.

Reply:
How would a user do this, as it is read only? Isn't this a configuration parameter, either?

Reply:
Thank you for making me rethink this. I'll try to clarify, then we can figure out what to do.

- The above channel contains the monitor id returned by the Zoneminder API. I made it a channel because the user likely will want to include the monitor id in their sitemap or HABpanel UI. As the monitor id is returned by the Zoneminder API, and cannot be changed by the user, I believe it should be readonly.

- As for the config parameter, when creating a monitor thing manually, or using a `.things` file, the user needs to specify the monitor id as part of defining the thing. When the monitor thing is created through discovery, it is set by the binding. Also, the monitor id the representation property.

So, what to do? I'm not sure how to resolve this without impacting one of the two points above. WDYT?

Reply:
Your arguments are valid. If I see correctly, the State of the Channel `id` is updated if the user changes the config parameter `monitorId`. I think it's good to go.

Reply:
Perfect. Thanks!